R/any_duplicated_linter.R

```diff
@@ -35,8 +35,7 @@
 #' @export
 any_duplicated_linter <- function() {
   any_duplicated_xpath <- "
-  //SYMBOL_FUNCTION_CALL[text() = 'any']
-    /parent::expr
+  parent::expr
```

Comment:
WDYT about `xml_find_function_calls()` landing on `//SYMBOL_FUNCTION_CALL/parent::expr`? It seems like most XPaths start there.

Reply:
It makes the docs harder to read and makes it more clunky to add additional conditions to the `SYMBOL_FUNCTION_CALL` node.
At the moment, migrating is "remove `//SYMBOL_FUNCTION_CALL[text() = foo]/` and run the xpath on `xml_find_function_calls("foo")`. If you need an additional condition, replace `//SYMBOL_FUNCTION_CALL[text() = 'foo' and cond]` by `self::SYMBOL_FUNCTION_CALL[cond]`.

Reply:
> makes it more clunky to add additional conditions to the SYMBOL_FUNCTION_CALL node.

A little bit, but it's only needed rarely, e.g. a handful of times in our codebase. More common was conditions on `parent::expr`, and from what I saw many of those could be migrated as well, e.g. instead of `[SYMBOL_FUNCTION_CALL]/parent::expr[following-sibling::expr[...]]` it is `[SYMBOL_FUNCTION_CALL/parent::expr]/parent::expr[expr[...]]`. Often that `parent::expr` is part of the XPath already so the result is even more concise.

> It makes the docs harder to read

Yes, a little, but I wonder if it's not more natural to land on `<expr>` nodes in general. They're the basic atoms of the AST, I know we used to write `//expr[SYMBOL_FUNCTION_CALL[...]]` by default until we realized that's inefficient & got used to `//SYMBOL_FUNCTION_CALL`.

BTW one might infer from the name `find_function_calls()` that we actually land on the _next_ outer `<expr>`, i.e. `//SYMBOL_FUNCTION_CALL/parent::expr/parent::expr`, since that node is where the full "call" is found. I don't have a better naming suggestion.

Reply:
Not sure how we want to proceed here. Maybe follow-up?
I want to avoid risking merge conflicts on this huge diff.

Reply:
OK, SG

Reply:
#2431